scalanet/src/io/iohk/scalanet/peergroup/SimplePeerGroup.scala

```diff
@@ -47,11 +47,15 @@ class SimplePeerGroup[A, AA, M](
   }
 
   override def server(): Observable[ServerEvent[A, M]] = {
-    underLyingPeerGroup.server().collectChannelCreated.map {
-      underlyingChannel: Channel[AA, Either[ControlMessage[A, AA], M]] =>
-        val reverseLookup: mutable.Map[AA, A] = routingTable.map(_.swap)
+    // FIXME pt1 addressing done as part of own proto
+    // FIXME pt2 somehow addressing as QoS option in underlying peer group allows fallback to underlying addressing?
+    val reverseLookup: mutable.Map[AA, A] = routingTable.map(_.swap)
```

Comment:
Can one underlying address (`AA` one) be assigned to many multicast ones (`A`)? What do we want to do in such case? Because now it seems that we are choosing random multicast address such that underlying incoming address is assigned to.

But I feel like that might be just one of issues possibly related to multicasting, I believe that we should create theoretic model for that and define contracts - we should create a separate task for that.

Reply:
During Slack conversation with Jeremy I understood that `routingTable` stores just mapping for single node addresses, not multicast ones, so there shouldn't be two keys with the same value (i.e. two `AA` underlying addresses with the same `A` one in `routingTable`, but they might exist in `multiCastTable`).